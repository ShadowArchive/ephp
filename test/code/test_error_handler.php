<?php

function handle_error($errno, $errstr, $errfile, $errline, $errcontext=array()) {
    static $a = 0;
    $errfile = basename($errfile);
    print "$errfile:$errline [$errno] $errstr\n";
    if ($a++ > 0) return false;
    return true;
}

function handle_error2($errno, $errstr, $errfile, $errline, $errcontext=array()) {
    print "wow!\n";
    return false;
}

print "older error handler (0) = " . set_error_handler("handle_error") . "\n";
print "older error handler (1) = " . set_error_handler("older_handle_error") . "\n";

# generate a warning...
sqrt("a");


print "older error handler (2) = " . set_error_handler("handle_error2") . "\n";
print "-----\n";
sqrt("a");

restore_error_handler();

print "-----\n";
sqrt("a");

restore_error_handler();

print "-----\n";
sqrt("a");
