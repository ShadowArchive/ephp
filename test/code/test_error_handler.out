older error handler (0) = 
test_error_handler.php:17 [2] set_error_handler() expects the argument (older_handle_error) to be a valid callback
older error handler (1) = 
test_error_handler.php:20 [2] sqrt() expects parameter 1 to be double, string given

Warning: sqrt() expects parameter 1 to be double, string given in {{CWD}}/test/code/test_error_handler.php on line 20
older error handler (2) = handle_error
-----
wow!

Warning: sqrt() expects parameter 1 to be double, string given in {{CWD}}/test/code/test_error_handler.php on line 25
-----
test_error_handler.php:30 [2] sqrt() expects parameter 1 to be double, string given

Warning: sqrt() expects parameter 1 to be double, string given in {{CWD}}/test/code/test_error_handler.php on line 30
-----

Warning: sqrt() expects parameter 1 to be double, string given in {{CWD}}/test/code/test_error_handler.php on line 35
